libdnf/hy-query.c

```diff
@@ -1049,10 +1049,16 @@ init_result(HyQuery q)
     Pool *pool = dnf_sack_get_pool(q->sack);
     Id solvid;
 
-    q->result = g_malloc0(sizeof(Map));
-    map_init(q->result, pool->nsolvables);
-    FOR_PKG_SOLVABLES(solvid)
-        map_set(q->result, solvid);
+    int sack_pool_nsolvables = dnf_sack_get_pool_nsolvables(q->sack);
+    if (sack_pool_nsolvables != 0 && sack_pool_nsolvables == pool->nsolvables)
```

Comment:
I don't think that this is reliable. Probably you want to store some hash and compare it if it didn't change.

Reply:
@ignatenkobrain Any solution here?

Reply:
@ignatenkobrain Code assumes that solvables are not reordered/replaced in pool after fill_sack().
I think If we break this rule then a lot of problems may occure. Eg. what about excludes/includes?